### DISCUSSION TITLE
How to pass additional data when navigating to dynamic routers?

### DISCUSSION THREAD

**Discussion by blasdfaa** [remix-run/remix#6016] [Q&A]
I use dynamic routing for the category page in the catalog. Right now the category page has this path - `/catalog/10`. My api server returns the category by ID.
Can I have category pages in remix like this - `/catalog/some-category-name`, but when I go to the page, pass in some way also the id to make a request to my api? But the user will just see the address - `/catalog/some-category-name`.

Using only `react-router-dom`, I could pass the `state` props to the `Link` component and get it via `useLocation`. 

How to implement this in remix?

**Comment by kiliman** [comment]
Not really. Since you're calling the loader on the server, it has to be in the URL.

So you can either pass the ID as a search param `/catalog/some-category-name?id=10`, or you'll need to look up the id from the category name param.

**Reply by blasdfaa** [reply]
> Not really. Since you're calling the loader on the server, it has to be in the URL.
> 
> So you can either pass the ID as a search param `/catalog/some-category-name?id=10`, or you'll need to look up the id from the category name param.

It seems to me that the option of passing query params will not be convenient, because categories have nested routes. `/catalog/some-category-name/neste-some-category-name` 😞
